tests/test.rs

```diff
@@ -299,9 +339,59 @@ fn test_integration() {
             assert_eq!(llvm_version, clang_version, "llvm-config ({:?}) and clang++ ({:?}) don't have the same major version", llvm_version, clang_version);
             make(path, "clang++");
             run(path);
-            check_equal_ade(&read_expected(path, "llvm", &llvm_version, "ade"), &run_grcov(path, true, "ade"));
-            check_equal_coveralls(&read_expected(path, "llvm", &llvm_version, "coveralls"), &run_grcov(path, true, "coveralls"), skip_branches);
+            check_equal_ade(&read_expected(path, "", "llvm", &llvm_version, "ade"),
+                            &run_grcov(vec![path], true, &PathBuf::from(""), "ade"));
+            check_equal_coveralls(&read_expected(path, "", "llvm", &llvm_version, "coveralls"),
+                                  &run_grcov(vec![path], true, path, "coveralls"), skip_branches);
+
             do_clean(path);
         }
     }
 }
+
+#[test]
+fn test_integration_zip() {
+    if cfg!(windows) {
+        println!("Integration tests still not supported under Windows.");
+        return;
+    }
+
+    println!("\nLLVM");
+    let path = &PathBuf::from("tests/basic");
+    let llvm_version = get_version("llvm-config");
+    let clang_version = get_version("clang++");
+    assert_eq!(llvm_version, clang_version, "llvm-config ({:?}) and clang++ ({:?}) don't have the same major version", llvm_version, clang_version);
+    make(path, "clang++");
+    run(path);
+
+    let gcno_zip_path = PathBuf::from("gcno.zip");
+    let gcda_zip_path = PathBuf::from("gcda.zip");
+    let gcda0_zip_path = PathBuf::from("gcda0.zip");
+    let gcda1_zip_path = PathBuf::from("gcda1.zip");
+
+    create_zip(&gcno_zip_path, path, "*.gcno");
+    create_zip(&gcda_zip_path, path, "*.gcda");
+    create_zip(&gcda0_zip_path, path, "");
+
+    let gcno_zip_path = path.join(gcno_zip_path);
+    let gcda_zip_path = path.join(gcda_zip_path);
+    let gcda0_zip_path = path.join(gcda0_zip_path);
+    let gcda1_zip_path = path.join(gcda1_zip_path);
+
+    // no gcda
+    check_equal_ade(&read_expected(path, "_no_gcda", "llvm", &llvm_version, "ade"),
```

Comment:
It might be better to use the `coveralls` format instead of `ade` for the test, so we also check the hit counts (they are not present in the `ade` format yet).